process/transactionLog/process.go

```diff
@@ -0,0 +1,122 @@
+package transactionLog
+
+import (
+	"encoding/hex"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var log = logger.GetOrCreate("process/transactionLog")
+
+// ArgTxLogProcessor defines the arguments needed for transaction logs processor
+type ArgTxLogProcessor struct {
+	Storer storage.Storer
+	Marshalizer marshal.Marshalizer
+}
+
+type txLogProcessor struct {
+	storer storage.Storer
+	marshalizer marshal.Marshalizer
+}
+
+// NewTxLogProcessor creates a transaction log processor capable of parsing logs from the VM
+//  and saving them into the injected storage
+func NewTxLogProcessor(args ArgTxLogProcessor) (process.TransactionLogProcessor, error) {
+	if check.IfNil(args.Storer) {
+		return nil, process.ErrNilStore
+	}
+	if check.IfNil(args.Marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+
+	return &txLogProcessor{
+		storer: args.Storer,
+		marshalizer: args.Marshalizer,
+	}, nil
+}
+
+// GetLog retreives a log generated by a transaction
+func (tlp *txLogProcessor) GetLog(txHash []byte) (data.LogHandler, error) {
+	txLogBuff, err := tlp.storer.Get(txHash)
+	if err != nil {
+		return nil, process.ErrLogNotFound
+	}
+
+	var txLog data.LogHandler
+	err = tlp.marshalizer.Unmarshal(txLog, txLogBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	return txLog, nil
+}
+
+// SaveLog takes the VM logs and saves them into the correct format in storage
+func (tlp *txLogProcessor) SaveLog(txHash []byte, tx data.TransactionHandler, vmLogs []*vmcommon.LogEntry) error {
+	if len(txHash) == 0 {
+		return process.ErrNilTxHash
+	}
+
+	if check.IfNil(tx) {
+		return process.ErrNilTransaction
+	}
+
+	if len(vmLogs) == 0 {
+		log.Debug("txLogProcessor.SaveLog()",
+			"txHash", hex.EncodeToString(txHash),
+			"message", "no vmLogs provided",
+		)
+
+		return nil
+	}
+
+	txLog := &transaction.Log{
+		Address: tx.GetRcvAddr(),
+	}
+
+	events := make([]*transaction.Event, len(vmLogs))
+	for i, vmLog := range vmLogs {
+		events[i] = tlp.computeEvent(vmLog)
+	}
+	txLog.Events = events
+
+	buff, err := tlp.marshalizer.Marshal(txLog)
+	if err != nil {
+		return err
+	}
+
+	return tlp.storer.Put(txHash, buff)
+}
+
+func (tlp *txLogProcessor) computeEvent(vmLog *vmcommon.LogEntry) *transaction.Event {
+	event := &transaction.Event {
+		Address: vmLog.Address,
+		Data: vmLog.Data,
+	}
+
+	topicsLen := len(vmLog.Topics)
+	if topicsLen == 0 {
+		return event
+	}
+	event.Identifier = vmLog.Topics[0]
```

Comment:
so identifier is the first topic, and the rest are the actual topics ? I think it would be better to change the vmcommon.LogEntry structure to make it easier to understand.

Reply:
Yes, this is by design from the rust -> wasm compiler. The event identifier from the annotation is added as the first topic (note that in c this is done manually because we directly call writeLog). I will change the LogEntry and writeLog then, makes more sense :)

Reply:
You have free hand to change the API to make more sense then what ethereum does :D